pkg/clients/clients.go

```diff
@@ -122,6 +117,7 @@ func (c *Clients) StartInformers(ctx context.Context) {
 	c.ConfigInformerFactory.Start(ctx.Done())
 	c.OperatorInformers.Start(ctx.Done())
 	c.ClusterExtensionClient.factory.Start(ctx.Done())
+	c.ClusterCatalogClient.factory.Start(ctx.Done())
```

Comment:
Have you seen any issues with this? We call `StartInformers` in `main.go` before the `ClusterCatalog` CRD exists on the cluster, and therefore before the API exists.

Is there some sort of retry logic that enables the ClusterCatalog informer to successfully start up after the API shows up?

Reply:
I have not seen any issues with this. I'm not sure if there is logic that retries under the hood. That being said, I've _personally_ only ever tested this in an environment where those CRDs had already been created. I do see that CI is green which seems to track that this is OK, although we don't have any `ClusterCatalog` resources we are managing yet.

Reply:
I'll do some additional digging to see if there is a need to take a different approach

Reply:
It does look like the `Reflector` for an informer will continue to List/Watch with exponential backoff on errors based on https://github.com/kubernetes/client-go/blob/c5146a9031d2c0dbe2a67dede62361ecf3d651bf/tools/cache/reflector.go#L309-L320

I think it's probably safe to leave as-is here with the intention that we eventually create the `ClusterCatalog` CRDs (I imagine fairly quickly after the informers are started) and the `Reflector` will eventually have a successful List/Watch operation.

If you would prefer we do something different, I'm open to exploring alternatives.

Reply:
Sounds good. Thanks for digging into it. I think we can leave as is, but we should keep this in mind to verify after this merges and https://github.com/openshift/operator-framework-catalogd/pull/81 is being tested.